lib/cucumber/runtime/ast_tree_walker.js

```diff
@@ -288,13 +292,15 @@ function AstTreeWalker(features, supportCodeLibrary, listeners, strictMode) {
     },
 
     processStep: function processStep(step, callback) {
-      if (self.isStepUndefined(step)) {
+      if(self.isDryRun()) {
```

Comment:
There should be a space between the if and the paren in keeping with the rest of the repo